### PR TITLE
Feature: Assign random avatar during parent registration (kids-1545 & kids-1547)

### PR DESCRIPTION
**Pull request by Daniela510**
- Move AvatarsCubit to getit
- Randomly assign avatar and keep track of them
- Display assigned avatar on us registration page
- Pass the assigned avatar through user registration

## Summary by CodeRabbit

- **New Features**
	- Introduced random avatar selection for users during registration.
	- Added a new `RandomAvatar` widget to display avatars.
	- Enhanced avatar management with new methods for assigning and retrieving avatars.
	- Updated user registration process to handle profile pictures.

- **Bug Fixes**
	- Improved state management for avatar selection screens by converting them to stateful widgets.

- **Chores**
	- Simplified provider setup for avatar selection routes.

### CODE REVIEW

File: lib/features/family/features/avatars/cubit/avatars_cubit.dart [hunk 1]
```diff
@@ -21,4 +23,42 @@ class AvatarsCubit extends Cubit<AvatarsState> {
...
+
+      emit(state.copyWith(assignedAvatars: updatedAvatars));
+    } else {
+      emit(state.copyWith(
+        assignedAvatars: [
+          ...state.assignedAvatars,
+          {key: randomAvatar}
+        ],
+      ));
+    }
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/avatars/cubit/avatars_cubit.dart:50]: this is set up for future use when we will have to assign avatars to a ist of users at once.

File: lib/features/family/features/avatars/cubit/avatars_cubit.dart [hunk 2]
```diff
@@ -21,4 +23,42 @@ class AvatarsCubit extends Cubit<AvatarsState> {
...
+  }
+
+  Avatar getAvatarByKey(String key) {
+    final assignment = state.assignedAvatars
+        .firstWhere((element) => element.containsKey(key), orElse: () => {});
+
+    return assignment[key] ?? const Avatar.empty();
+  }
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/avatars/cubit/avatars_cubit.dart:58]: key is not index, this does not work.

File: lib/features/registration/bloc/registration_bloc.dart [hunk 3]
```diff
@@ -78,6 +78,7 @@ class RegistrationBloc extends Bloc<RegistrationEvent, RegistrationState> {
         postalCode: event.postalCode,
         accountNumber: event.accountNumber,
         sortCode: event.sortCode,
+        profilePicture: event.profilePicture,
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/bloc/registration_bloc.dart:81]: false.

File: lib/features/registration/widgets/random_avatar.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  final void Function() onClick;
+  @override
+  State<RandomAvatar> createState() => _RandomAvatarState();
+}
+
+class _RandomAvatarState extends State<RandomAvatar> {
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    getIt<AvatarsCubit>().assignRandomAvatarUrl(widget.id);
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/registration/widgets/random_avatar.dart:25]: good

**Review by TammiLion — APPROVED** [review]

**Pull request merged**